app/controllers/api/v1/reservations_controller.rb

```diff
@@ -0,0 +1,51 @@
+class Api::V1::ReservationsController < ApplicationController
+  before_action :set_reservation, only: %i[show update destroy]
+
+  # GET /reservations
+  def index
+    @reservations = current_user.reservations.all
+
+    render json: @reservations
```

Comment:
Here, while listing the json for all reservations you can include their related fitness_activity using the Active Model Serializer.
i.e.` render json: @reservation, status: 200, include: %i[fitness_activity]`